### PR TITLE
feature/FPHS-702

### PR DESCRIPTION
**Pull request by dephillipsmichael**
Added new helper method to tasks reminder manager that can determine what a "new" task is per 3 conditional qualifications, see story.  Also added unit tests for all important methods, which most of this code is.

### CODE REVIEW

File: APCAppCore/APCAppCore/Library/Objects/APCTasksReminderManager.h [hunk 1]
```diff
@@ -81,4 +82,22 @@ static NSUInteger const kAPCTaskReminderDayOfWeekSaturday    = 7;
...
+ *
+ * @param oldTaskGroups the result from reading cache BEFORE sycning from the web
+ * @param newTaskGroups the result from reading cache AFTER  sycning from the web
+ *
+ * @return an array of task group objects that are considered "new" per the 3 conditions above
+ */
++ (NSArray*)findNewTasksFromOld:(NSArray*)oldTaskGroups
+                toNewTaskGroups:(NSArray*)newTaskGroups;
+
++ (BOOL) isMoreThan3DaysOld:(APCTask*) task;
```
Comment (EricSiegNW) [inline APCAppCore/APCAppCore/Library/Objects/APCTasksReminderManager.h]: Can it be more than X days old, with an age set in the app delegate? Other apps might need different delays.

File: APCAppCore/APCAppCoreTests/Reminder Tests/APCTasksReminderManagerTests.m [hunk 2]
```diff
@@ -297,4 +303,211 @@ - (void) testMPowerDaily
...
+    NSArray* newReminderTasks = [APCTasksReminderManager findNewTasksFromOld:oldTaskGroups toNewTaskGroups:newTaskGroups];
+    
+    XCTAssertTrue(newReminderTasks != nil);
+    XCTAssertTrue(newReminderTasks.count > 0);
+    XCTAssertEqual(((APCTask*)newReminderTasks[0]).taskID, @"qow");
+}
+
+- (void) testSurveyRecurringAdded
+{
+    NSArray* oldTaskGroups = @[[self taskGroupWithID:@"qow"
```
Comment (EricSiegNW) [inline APCAppCore/APCAppCoreTests/Reminder Tests/APCTasksReminderManagerTests.m]: this seems to be bleeding FPHS specific stuff into appcore

Comment (dephillipsmichael) [inline APCAppCore/APCAppCoreTests/Reminder Tests/APCTasksReminderManagerTests.m]: you mean the specific task ID or the unit tests in general?

File: APCAppCore/APCAppCore/Library/Objects/APCTasksReminderManager.h (hunk 1 above)
Comment (dephillipsmichael) [inline APCAppCore/APCAppCore/Library/Objects/APCTasksReminderManager.h]: good idea

File: APCAppCore/APCAppCoreTests/Reminder Tests/APCTasksReminderManagerTests.m (hunk 2 above)
Comment (EricSiegNW) [inline APCAppCore/APCAppCoreTests/Reminder Tests/APCTasksReminderManagerTests.m]: taskID. the "qow" gives the impression (to me) that its FPHS specific. If you are just picking any taskID, should probably be a more generic one.

Comment (dephillipsmichael) [inline APCAppCore/APCAppCoreTests/Reminder Tests/APCTasksReminderManagerTests.m]: good point, ill make it "survey"

File: APCAppCore/APCAppCore/Library/Objects/APCTasksReminderManager.h [hunk 3]
```diff
@@ -92,12 +92,15 @@ static NSUInteger const kAPCTaskReminderDayOfWeekSaturday    = 7;
...
  * @param newTaskGroups the result from reading cache AFTER  sycning from the web
+ * @param daysIncompleteToRemind the number of days that an activity can remain incomplete before it is considered "new"
  *
  * @return an array of task group objects that are considered "new" per the 3 conditions above
  */
-+ (NSArray*)findNewTasksFromOld:(NSArray*)oldTaskGroups
-                toNewTaskGroups:(NSArray*)newTaskGroups;
++ (NSArray*)findNewTasksFromOld:(NSArray*)  oldTaskGroups
+                toNewTaskGroups:(NSArray*)  newTaskGroups
+   withAcitivtyReminderDuration:(NSInteger) daysIncomplete;
```
Comment (EricSiegNW) [inline APCAppCore/APCAppCore/Library/Objects/APCTasksReminderManager.h]: spelling typo

Comment (dephillipsmichael) [inline APCAppCore/APCAppCore/Library/Objects/APCTasksReminderManager.h]: nooooooo

File: APCAppCore/APCAppCore/Library/Objects/APCTasksReminderManager.m [hunk 4]
```diff
@@ -672,4 +672,67 @@ -(BOOL)includeTaskInReminder:(APCTaskReminder *)taskReminder{
...
+        oldTaskCompletedStatusById[taskGroup.task.taskID] = @(taskGroup.isFullyCompleted);
+        oldTaskById[taskGroup.task.taskID]                = taskGroup.task;
+    }
+    
+    // Right now there are a few conditions for determining if a task should be reminded about...
+    // 1) for any task, if the taskId was not in the old list, and is now in the new list
+    // 2) for any task, if the task was completed in old, but is incomplete in new
+    // 3) for activities, if the task is incomplete, old task is less than 3 days old, and new
+    //    task is more than 3 days old
+    NSMutableArray* newTasks = [@[] mutableCopy];
```
Comment (Erin-Mounts) [inline APCAppCore/APCAppCore/Library/Objects/APCTasksReminderManager.m]: [NSMutableArray array] to create an empty mutable array--no need to create a static empty NSArray and then make a mutable copy

File: APCAppCore/APCAppCore/Library/Objects/APCTasksReminderManager.m [hunk 5]
```diff
@@ -672,4 +672,67 @@ -(BOOL)includeTaskInReminder:(APCTaskReminder *)taskReminder{
...
+    for (APCTaskGroup* taskGroup in newTaskGroups)
+    {
```
Comment (Erin-Mounts) [inline APCAppCore/APCAppCore/Library/Objects/APCTasksReminderManager.m]: editing existing code, so follow existing formatting conventions please ( '{' on same line as end of if/for/else clause, '}' on same line as else )

Comment (dephillipsmichael) [inline APCAppCore/APCAppCore/Library/Objects/APCTasksReminderManager.m]: My bad, I'm usually pretty good about that.  Each library has different conventions, so I forget which one I'm editing sometimes.  Will fix.

**Comment by dephillipsmichael** [comment]
@Erin-Mounts fixed this for you.  were there any other blockers?

**Comment by Erin-Mounts** [comment]
looks good

**Pull request merged**